ksql-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalysis.java

```diff
@@ -66,4 +68,29 @@ public Expression getHavingExpression() {
   public void setHavingExpression(Expression havingExpression) {
     this.havingExpression = havingExpression;
   }
+
+  public void addAggregateFunctionArgument(final Expression argument) {
+    aggregateFunctionArguments.add(argument);
+  }
+
+  public void addFunction(final FunctionCall functionCall) {
+    functionList.add(functionCall);
+  }
+
+  public boolean hasRequiredColumn(final String column) {
+    return requiredColumnsMap.containsKey(column);
+  }
+
+  public void addRequiredColumn(final String name, final DereferenceExpression node) {
```

Comment:
Although currently we just support `DereferenceExpression` in aggregate columns we will support other expressions too so this should be `Expression` instead of `DereferenceExpression`.